zipline/pipeline/loaders/_13d_filings.py

```diff
@@ -0,0 +1,54 @@
+"""
+Reference implementation for 13d filings loaders.
+"""
+
+from zipline.pipeline.common import (
+    DISCLOSURE_DATE,
+    PERCENT_SHARES,
+    NUM_SHARES
+)
+from zipline.pipeline.data import _13DFilings
+from zipline.pipeline.loaders.events import EventsLoader
+from zipline.utils.memoize import lazyval
+
+
+class _13DFilingsLoader(EventsLoader):
+    """
+    Reference loader for
+    :class:`zipline.pipeline.data._13DFilings`.
+
+    events_by_sid: dict[sid -> pd.DataFrame(knowledge date,
+    disclosure date, percent shares, number of shares)]
+
+    """
+    expected_cols = frozenset([DISCLOSURE_DATE,
+                               PERCENT_SHARES,
+                               NUM_SHARES])
+    event_date_col = DISCLOSURE_DATE
```

Comment:
Yeah, this was introduced in a very recent PR to zipline. Each event dataset now needs to explicitly define an event_date_col which implies certain behavior for other parts of the code.

> On Apr 28, 2016, at 11:37 AM, Kathryn Glowinski <notifications@github.com> wrote:
> 
> In zipline/pipeline/loaders/_13d_filings.py <https://github.com/quantopian/zipline/pull/1106#discussion_r61450012>:
> 
> > +from zipline.utils.memoize import lazyval
> > +
> > +
> > +class _13DFilingsLoader(EventsLoader):
> > +    """
> > +    Reference loader for
> > +    :class:`zipline.pipeline.data._13DFilings`.
> > +
> > +    events_by_sid: dict[sid -> pd.DataFrame(knowledge date,
> > +    disclosure date, percent shares, number of shares)]
> > +
> > +    """
> > +    expected_cols = frozenset([DISCLOSURE_DATE,
> > +                               PERCENT_SHARES,
> > +                               NUM_SHARES])
> > +    event_date_col = DISCLOSURE_DATE
> What does having this column as the event_date_col do? Was this not something that was needed before that we recently added?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/quantopian/zipline/pull/1106/files/9e3c83a4aabd164debb69fc1a9b3273f00bc6042#r61450012>